api/src/main/java/org/openmrs/module/attachments/obs/ImageAttachmentHandler.java

```diff
@@ -14,94 +14,92 @@
-	               + "FILE='" + complexData.getTitle() + "'.", e);
-		}
-
-		return new ValueComplex(complexData.getInstructions(), complexData.getMimeType(), savedFileName);
-	}
+
+    public final static int THUMBNAIL_MAX_HEIGHT = 200;
+
+    public final static int THUMBNAIL_MAX_WIDTH = THUMBNAIL_MAX_HEIGHT;
+
+    public ImageAttachmentHandler() {
+        super();
+    }
+
+    @Override
+    protected void setParentComplexObsHandler() {
+        setParent(new ImageHandler());
+    }
+
+    @Override
+    protected ComplexData readComplexData(Obs obs, ValueComplex valueComplex, String view) {
+
+        String fileName = valueComplex.getFileName();
+        if (view.equals(AttachmentsConstants.ATT_VIEW_THUMBNAIL)
+                && !StringUtils.endsWith(FilenameUtils.removeExtension(fileName), NO_THUMBNAIL_SUFFIX)) {
+            fileName = buildThumbnailFileName(fileName);
+        }
+
+        // We invoke the parent to inherit from the file reading routines.
+        Obs tmpObs = new Obs();
+        tmpObs.setValueComplex(fileName); // Temp obs used as a safety
+        tmpObs = getParent().getObs(tmpObs, AttachmentsConstants.IMAGE_HANDLER_VIEW); // ImageHandler doesn't handle several views
+        ComplexData complexData = tmpObs.getComplexData();
+
+        // Then we build our own custom complex data
+        return getComplexDataHelper().build(valueComplex.getInstructions(), complexData.getTitle(), complexData.getData(),
+                valueComplex.getMimeType()).asComplexData();
+    }
+
+    @Override
+    protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData) {
+
+        // We use a temp obs whose complex data points to the file names
+        String fileName = complexData.getTitle();
+        boolean isThumbNailPurged = true;
+        Obs tmpObs = new Obs();
+
+        if (!StringUtils.endsWith(FilenameUtils.removeExtension(fileName), NO_THUMBNAIL_SUFFIX)) {
+            String thumbnailFileName = buildThumbnailFileName(fileName);
+            tmpObs.setValueComplex(thumbnailFileName);
+            isThumbNailPurged = getParent().purgeComplexData(tmpObs);
+        }
+
+        tmpObs.setValueComplex(fileName);
+        boolean isImagePurged = getParent().purgeComplexData(tmpObs);
+
+        return isThumbNailPurged && isImagePurged;
+    }
+
+    @Override
+    protected ValueComplex saveComplexData(Obs obs, AttachmentComplexData complexData) {
+        int imageHeight = Integer.MAX_VALUE;
+        int imageWidth = Integer.MAX_VALUE;
+        String savedFileName = null;
+
+        // We invoke the parent to inherit from the file saving routines.
+        obs = getParent().saveObs(obs);
+
+        File savedFile = AbstractHandler.getComplexDataFile(obs);
+
+        // Get image dimensions
+        try {
+            BufferedImage image = ImageIO.read(savedFile);
+            imageHeight = image.getHeight();
+            imageWidth = image.getWidth();
+        } catch (IOException e) {
```

Comment:
I have been thinking here. If the dimensions could not be determined, then you should not throw an exception but let the code go on with maximum dimensions. Then after that if the thumbnail generation fails, _then_ so be it. So basically just:
```java

try {
  BufferedImage image = ImageIO.read(savedFile);
  imageHeight = image.getHeight();
  imageWidth = image.getWidth();
} catch (IOException e) {
  log.warn("The dimensions of image 'savedFileName' could not be determined, continuing with generating thumbnail.")  
}
```